flow/record/base.py

```diff
@@ -723,67 +802,89 @@ def open_path(path, mode, clobber=True):
             fp = io.open(path, mode)
         # check if we are reading a compressed stream
         if not out and binary:
-            if not hasattr(fp, "peek"):
-                fp = Peekable(fp)
-            peek_data = fp.peek(4)
-            if peek_data[:2] == GZIP_MAGIC:
-                fp = gzip.GzipFile(fileobj=fp, mode=mode)
-            elif HAS_BZ2 and peek_data[:3] == BZ2_MAGIC:
-                fp = bz2.BZ2File(fp, mode=mode)
-            elif HAS_LZ4 and peek_data[:4] == LZ4_MAGIC:
-                fp = lz4.open(fp, mode=mode)
-            elif HAS_ZSTD and peek_data[:4] == ZSTD_MAGIC:
-                dctx = zstd.ZstdDecompressor()
-                fp = dctx.stream_reader(fp)
+            fp = open_stream(fp, mode)
     return fp
 
 
-def RecordAdapter(url, out, selector=None, clobber=True, **kwargs):
-    url = str(url or "")
-
+def RecordAdapter(
+    url: str = None, out: bool = False, selector=None, clobber: bool = True, fileobj: BinaryIO = None, **kwargs
```

Comment:
```suggestion
    url: str = None, out: bool = False, selector: Optional[str] = None, clobber: bool = True, fileobj: Optional[BinaryIO] = None, **kwargs
```
I believe `selector` is a `str`.